WalletWasabi/Tor/TcpClientConnector.cs

```diff
@@ -14,24 +10,32 @@ public static class TcpClientConnector
 	/// <summary>
 	/// Connects to end point using a TCP client.
 	/// </summary>
-	/// <param name="builder">Runs in between creation of TCP client and connection to the end point.</param>
-	public static async Task<TcpClient> ConnectAsync(EndPoint endPoint, CancellationToken cancel, Action<TcpClient>? builder = null)
+	public static async Task<TcpClient> ConnectAsync(EndPoint endPoint, CancellationToken cancellationToken)
 	{
-		TcpClient tcpClient = new(endPoint.AddressFamily);
-		builder?.Invoke(tcpClient);
-		switch (endPoint)
-		{
-			case DnsEndPoint dnsEndPoint:
-				await tcpClient.ConnectAsync(dnsEndPoint.Host, dnsEndPoint.Port, cancel).ConfigureAwait(false);
-				break;
+		TcpClient? client = null;
 
-			case IPEndPoint ipEndPoint:
-				await tcpClient.ConnectAsync(ipEndPoint, cancel).ConfigureAwait(false);
-				break;
+		try
+		{
+			client = new(endPoint.AddressFamily);
```

Comment:
This will crash when a DnsEndpoint is provided

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/8897/commits/ad6d054ac9c0543cc10434dff1f2064eb3dfecba